xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/web/ExportAction.java

```diff
@@ -195,11 +332,11 @@ private String exportHTML(XWikiContext context) throws XWikiException, IOExcepti
                     List<String> docsNames = dbQuery.setWiki(wikiName).bindValues(params).execute();
 
                     for (String docName : docsNames) {
-                        String pageReference = wikiName + XWikiDocument.DB_SPACE_SEP + docName;
-                        if (context.getWiki().getRightService().hasAccessLevel(
-                            "view", context.getUser(), pageReference, context))
-                        {
-                            pageList.add(resolver.resolve(pageReference));
+                        String pageReferenceStr = wikiName + XWikiDocument.DB_SPACE_SEP + docName;
```

Comment:
Better pass a WikiReference as second parameter of the resolver (should always avoid directly manipulate reference syntax).